File: objc-demo/ChatDemo-UI3.0/Code/Utilities/Category/NSObject+EMAlertView.m
/************************************************************
 *  * Hyphenate
 * __________________
 * Copyright (C) 2016 Hyphenate Inc. All rights reserved.
 *
 * NOTICE: All information contained herein is, and remains
 * the property of Hyphenate Inc.
 */
#import "NSObject+EMAlertView.h"

@implementation NSObject (EMAlertView)

- (void)showAlertWithMessage:(NSString *)msg {
    UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:@""
                                                        message:msg
                                                       delegate:nil
                                              cancelButtonTitle:NSLocalizedString(@"common.ok", @"OK")
                                              otherButtonTitles:nil, nil];
    [alertView show];
}


- (void)showAlertWithMessage:(NSString *)msg delegate:(id)delegate {
    UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:@""
                                                        message:msg
                                                       delegate:delegate
                                              cancelButtonTitle:NSLocalizedString(@"common.ok", @"OK")
                                              otherButtonTitles:nil, nil];
    [alertView show];
}

@end

File: swift-demo/Hyphenate-Demo-Swift/Code/Utilities/Category/NSObject+EMAlertView.swift
//
//  NSObject+EMAlertView.swift
//  Hyphenate-Demo-Swift
//
//  Created by 杜洁鹏 on 2017/6/27.
//  Copyright © 2017年 杜洁鹏. All rights reserved.
//

import Foundation
import UIKit

extension NSObject {
    @nonobjc
    func showAlert(_ message: String){
        showAlert(message, nil)
    }
    
    @nonobjc
    func showAlert(_ message: String, _ delegate: UIAlertViewDelegate?) {
        showAlert(nil, message, delegate)
    }
 
    @nonobjc
    func showAlert(_ title: String, _ message: String) {
        showAlert(title, message, nil)
    }
    
    @nonobjc
    func showAlert(_ title: String?, _ message: String, _ delegate: UIAlertViewDelegate?){
        let alertView = UIAlertView.init(title: title, message: message, delegate: delegate, cancelButtonTitle: "OK")
        alertView.show()
    }
}
